herddb-core/src/main/java/herddb/codec/RecordSerializer.java

```diff
@@ -282,12 +304,13 @@ public static void skipTypeAndValue(ByteArrayCursor dii) throws IOException {
 
     public static byte[] serialize(Object v, int type) {
         if (v == null) {
-            return null;
+            throw new IllegalArgumentException("Object to be serialized cannot be null");
```

Comment:
we cannot change this behevious
and NULL is a valid value to be serialized

Reply:
Done. I reverted this.